### PR TITLE
Remove content link

### PR DESCRIPTION
**Pull request by JoseAlba**
## Thanks for starting a pull request on Material Components!
Removes unnecessary content link for each dev doc.

#### Don't forget:
- [x] Are you sure you mean this repo? This repository is a reference to the [main Flutter repository](https://github.com/flutter/flutter). Code-level issues should be created there.
- [] Link to GitHub issues it solves. ```closes #1234```
- [x] Sign the CLA bot. You can do this once the pull request is opened.

Be sure to read the [Flutter guide on contributing](https://github.com/flutter/flutter/blob/master/CONTRIBUTING.md).

### Related issues
 - Type `#` and the issue number or `#` and select the from the inline drop down.

### CODE REVIEW

File: docs/components/checkboxes.md [hunk 1]
```diff
@@ -21,11 +21,16 @@ Use checkboxes to:
 
 ## Contents
 
-- [Contents](#contents)
-- [Using checkboxes](#using-checkboxes)
-- [Checkboxes](#checkboxes)
-- [Checkbox key properties](#checkbox-key-properties)
-- [Theming](#theming)
+- [Selection controls: checkboxes](#selection-controls-checkboxes)
```
Comment (guidezpl) [inline docs/components/checkboxes.md:24]: Should remove the first and second links here too, no?

Comment (rami-a) [inline docs/components/checkboxes.md:24]: This change looks incorrect. I think we are only supposed to include top level headers, and `contents` is still in here.

Comment (JoseAlba) [reply docs/components/checkboxes.md]: Done

Comment (JoseAlba) [reply docs/components/checkboxes.md]: Done

**Pull request merged**